### PR TITLE
feat(aws-amplify-react-native): Use SecureRandom instead of Random for better security.

### PR DESCRIPTION
**Pull request by vikas5914**
_Issue #5272, 

_Description of changes:_

Replace `java.util.Random` with `import java.security.SecureRandom` for better security. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/pushnotification/android/src/main/java/com/amazonaws/amplify/pushnotification/RNPushNotificationMessagingService.java [hunk 1]
```diff
@@ -107,7 +107,7 @@ private void handleFCMMessagePush(ReactApplicationContext context, Bundle bundle
     private void sendNotification(ReactApplicationContext context, Bundle bundle, Boolean isForeground) {
         // If notification ID is not provided by the user for push notification, generate one at random
         if (bundle.getString("id") == null) {
-            Random randomNumberGenerator = new Random(System.currentTimeMillis());
+            SecureRandom randomNumberGenerator = new SecureRandom();
```
Comment (amhinson) [inline packages/pushnotification/android/src/main/java/com/amazonaws/amplify/pushnotification/RNPushNotificationMessagingService.java:110]: I'm wondering if we should still seed this, just for consistency. Maybe something like:

```java
SecureRandom randomNumberGenerator = new SecureRandom();
randomNumberGenerator.setSeed(System.currentTimeMillis());
```
Reference: https://docs.oracle.com/javase/8/docs/api/java/security/SecureRandom.html#setSeed-long-

Comment (vikas5914) [reply packages/pushnotification/android/src/main/java/com/amazonaws/amplify/pushnotification/RNPushNotificationMessagingService.java:110]: SecureRandom requires a long seed and System.currentTimeMillis() output is not long.

Comment (amhinson) [reply packages/pushnotification/android/src/main/java/com/amazonaws/amplify/pushnotification/RNPushNotificationMessagingService.java:110]: Oh interesting. Based on the [docs](https://docs.oracle.com/javase/8/docs/api/java/util/Random.html), `Random` constructor uses a `long seed`, but `SecureRandom` constructor uses a `byte[] seed`. However you can pass a `long seed` to `setSeed`.

Did you try using the `setSeed` method for `SecureRandom`?

Comment (vikas5914) [reply packages/pushnotification/android/src/main/java/com/amazonaws/amplify/pushnotification/RNPushNotificationMessagingService.java:110]: with setSeed , it working.

**Pull request merged**